tests/integration/bond_test.py

```diff
@@ -403,3 +406,77 @@ def test_change_bond_option_miimon(bond99_with_2_slaves):
     bond_options["miimon"] = "200"
     libnmstate.apply(desired_state)
     assertlib.assert_state(desired_state)
+
+
+def test_create_bond_without_mode():
+    with bond_interface(name=BOND99, slaves=[], create=False) as state:
+        state[Interface.KEY][0][Bond.CONFIG_SUBTREE].pod(Bond.MODE)
+        with pytest.raises(NmstateValueError):
+            libnmstate.apply(state)
+
+
+def test_bond_mac_restriction_without_mac_in_desire(eth1_up, eth2_up):
+    with bond_interface(
+        name=BOND99,
+        slaves=[ETH1, ETH2],
+        extra_iface_state={
+            Bond.CONFIG_SUBTREE: {
+                Bond.MODE: BondMode.ACTIVE_BACKUP,
+                Bond.OPTIONS_SUBTREE: {"fail_over_mac": "active"},
+            },
+        },
+    ) as state:
+        assertlib.assert_state_match(state)
+
+
+def test_bond_mac_restriction_with_mac_in_desire(eth1_up, eth2_up):
```

Comment:
IMO, this is being tested in the test_mac_restriction_with_mac_in_desire validator test. So there is no need to test this here.

Reply:
That is unit test, this is integration test.
Unit test is testing the functionality of single component.
Integration test is testing the user case actually works with all components.